materialize-snowflake/snowflake.go

```diff
@@ -446,9 +454,9 @@ func (d *transactor) Load(it *m.LoadIterator, loaded func(int, json.RawMessage)
 }
 
 type checkpointItem struct {
-	Table    string
-	Query    string
-	ToDelete string
```

Comment:
With the way we are using merge patches, I don't think this will ever completely remove the old `ToDelete` field. I think it would technically still work to always have that field in legacy checkpoints since we aren't doing a strict unmarshalling the forbids unknown fields, but I'd feel better if we explicitly deleted the field from checkpoints so it doesn't cause problems down the road. 

We should be able to use something like this:
```go
type checkpointItem struct {
	Table     string
	Query     string
	StagedDir string
	ToDelete  *string `json:",omitempty"`
}
```

And then for the migration part, set `ToDelete` to `nil` so that it gets marshalled as `null` in the merge patch and fully deleted from the checkpoint.

Reply:
I'll actually walk this back - since we are emitting checkpoints at post-ack that have `checkpointItem: nil`, as long as one of _those_ gets emitted, then the checkpoint will be cleared of its old field. And it's quite reasonable to assume that those post-ack checkpoints will be emitted in a timely fashion, since it always does one at startup.